autokeras/tuner.py

```diff
@@ -72,27 +68,71 @@ def load_trial(self, trial):
         path = os.path.join(trial.directory, filename)
         self.hypermodel.load_preprocessors(path)
 
-    @staticmethod
-    def add_earlystopping_callback(callbacks):
-        if not callbacks:
-            callbacks = []
-
-        try:
-            callbacks = copy.deepcopy(callbacks)
-        except:
-            raise ValueError(
-                'All callbacks used during a search '
-                'should be deep-copyable (since they are '
-                'reused across executions). '
-                'It is not possible to do `copy.deepcopy(%s)`' %
-                (callbacks,))
-
-        if not [callback for callback in callbacks
-                if isinstance(callback, tf.keras.callbacks.EarlyStopping)]:
-            # The patience is set to 30 based on human experience.
-            callbacks.append(tf.keras.callbacks.EarlyStopping(patience=30))
-
-        return callbacks
+    def search(self, *fit_args, **fit_kwargs):
+        # Format the arguments.
+        fit_kwargs.update(
+            dict(zip(inspect.getfullargspec(tf.keras.Model.fit).args, fit_args)))
+        fit_args = []
+        fit_kwargs.setdefault('callbacks', [])
+        if fit_kwargs['callbacks'] is None:
+            fit_kwargs['callbacks'] = []
+        fit_kwargs.setdefault('epochs', None)
+
+        # Insert early-stopping permanently when epochs is None.
+        callbacks = fit_kwargs['callbacks']
+        if fit_kwargs['epochs'] is None:
+            fit_kwargs['epochs'] = 1000
+            if not any([isinstance(callback, tf.keras.callbacks.EarlyStopping)
+                        for callback in callbacks]):
+                callbacks = callbacks + [
+                    tf.keras.callbacks.EarlyStopping(patience=30)]
```

Comment:
30 seems somewhat excessive, it guarantees overfitting. ~10 seems more reasonable, no?